lib/Drivers/EulerForward/Driver.cpp

```diff
@@ -1,77 +1,79 @@
 #include <iostream>
 
-namespace marco::runtime
-{
-  EulerForward::EulerForward(Simulation* simulation)
-      : Driver(simulation)
-  {
-  }
+namespace marco::runtime {
+EulerForward::EulerForward(Simulation *simulation) : Driver(simulation) {}
 
 #ifdef CLI_ENABLE
-  std::unique_ptr<cli::Category> EulerForward::getCLIOptions()
-  {
-    return std::make_unique<eulerforward::CommandLineOptions>();
-  }
+std::unique_ptr<cli::Category> EulerForward::getCLIOptions() {
+  return std::make_unique<eulerforward::CommandLineOptions>();
+}
 #endif // CLI_ENABLE
 
-  int EulerForward::run()
-  {
-    if (marco::runtime::simulation::getOptions().debug) {
-      std::cerr << "[Euler Forward] Starting simulation" << std::endl;
-    }
-
-    double time;
+int EulerForward::run() {
+  if (marco::runtime::simulation::getOptions().debug) {
+    std::cerr << "[Euler Forward] Starting simulation" << std::endl;
+  }
 
-    do {
-      // Compute the next values of the state variables.
-      if (marco::runtime::simulation::getOptions().debug) {
-        std::cerr << "[Euler Forward] Updating state variables" << std::endl;
-      }
+  double time;
+  std::size_t iterationStep = 0;
 
-      EULER_FORWARD_PROFILER_STATEVAR_START;
-      updateStateVariables(eulerforward::getOptions().timeStep);
-      EULER_FORWARD_PROFILER_STATEVAR_STOP;
+  do {
+    iterationStep++;
 
-      // Move to the next step.
-      if (marco::runtime::simulation::getOptions().debug) {
-        std::cerr << "[Euler Forward] Updating time and non-state variables" << std::endl;
-      }
+    // Compute the next values of the state variables.
+    if (marco::runtime::simulation::getOptions().debug) {
+      std::cerr << "[Euler Forward] Updating state variables" << std::endl;
+    }
 
-      EULER_FORWARD_PROFILER_NONSTATEVAR_START;
-      time = getTime() + eulerforward::getOptions().timeStep;
-      setTime(time);
+    EULER_FORWARD_PROFILER_STATEVAR_START;
+    updateStateVariables(eulerforward::getOptions().timeStep);
+    EULER_FORWARD_PROFILER_STATEVAR_STOP;
 
-      updateNonStateVariables();
-      EULER_FORWARD_PROFILER_NONSTATEVAR_STOP;
+    // Move to the next step.
+    if (marco::runtime::simulation::getOptions().debug) {
+      std::cerr << "[Euler Forward] Updating time and non-state variables"
+                << std::endl;
+    }
 
-      if (marco::runtime::simulation::getOptions().debug) {
-        std::cerr << "[Euler Forward] Printing values" << std::endl;
-      }
+    EULER_FORWARD_PROFILER_NONSTATEVAR_START;
+    time = getTime() + eulerforward::getOptions().timeStep;
+    setTime(time);
 
-      // Print the values.
-      getSimulation()->getPrinter()->printValues();
-    } while (std::abs(simulation::getOptions().endTime - time) >=
-             eulerforward::getOptions().timeStep);
+    updateNonStateVariables();
+    EULER_FORWARD_PROFILER_NONSTATEVAR_STOP;
 
     if (marco::runtime::simulation::getOptions().debug) {
```

Comment:
This should be done only if the snapshot steps condition holds.

Reply:
Pushing soon

Reply:
The macros / assertions -- introduced as sequence points?